src/Idris/Desugar.idr

```diff
@@ -735,14 +735,15 @@ mutual
       = do tm' <- desugarDo side ps ns tm
            rest' <- expandDo side ps topfc ns rest
            pure $ seqFun fc ns tm' rest'
-  expandDo side ps topfc ns (DoBind fc nameFC n tm :: rest)
+  expandDo side ps topfc ns (DoBind fc nameFC n rig ty tm :: rest)
       = do tm' <- desugarDo side ps ns tm
            rest' <- expandDo side ps topfc ns rest
            whenJust (isConcreteFC nameFC) $ \nfc => addSemanticDecorations [(nfc, Bound, Just n)]
+           ty' <- maybe (pure $ Implicit (virtualiseFC fc) False)
+                        (\ty => desugarDo side ps ns ty) ty
```

Comment:
Isn't it bizarre that we are using `desugarDo` to elaborate the type?
Does it even make sense to have an effectful function returning a type here?

Reply:
Yeah, that struck me as odd, but I followed the pattern of `DoLet` and `DoLetPat`. It looks like they were changed from a full `desugar` to `desugarDo` in  ee063a5 to propagate the `do` namespace.

Reply:
Right, I think that should be reverted.
Unless @Russoul has a motivation for it.

Reply:
I'll let you guys sort out what is wanted here. It looks like `desugar` simply calls `desugarDo` with a namespace of `Nothing`.

Reply:
Yeah I think the function name threw me off, but either looks fine. The difference seems to be what namespace ends up on comprehensions, which I wouldn't expect to appear in types.

What did you want to revert? Is it just changing the other`ty` elaborations to `desugar` or more of #1700?